samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js

```diff
@@ -0,0 +1,169 @@
+     * Currently, this expects a booking request, like "book me a flight from Paris to Berlin on march 22"
+     * Note that the sample LUIS model will only recognize Paris, Berlin, New York and London as airport cities.
+     */
+    async introStep(stepContext) {
+        if (!this.luisRecognizer.isConfigured) {
+            const messageText = 'NOTE: LUIS is not configured. To enable all capabilities, add `LuisAppId`, `LuisAPIKey` and `LuisAPIHostName` to the .env file.';
+            await stepContext.context.sendActivity(messageText, null, InputHints.IgnoringInput);
+            return await stepContext.next();
+        }
+
+        const messageText = stepContext.options.restartMsg ? stepContext.options.restartMsg : 'What can I help you with today?\nSay something like "Book a flight from Paris to Berlin on March 22, 2020"';
+        const promptMessage = MessageFactory.text(messageText, messageText, InputHints.ExpectingInput);
+        return await stepContext.prompt('TextPrompt', { prompt: promptMessage });
+    }
+
+    /**
+     * Second step in the waterfall.  This will use LUIS to attempt to extract the origin, destination and travel dates.
+     * Then, it hands off to the bookingDialog child dialog to collect any remaining details.
+     */
+    async actStep(stepContext) {
+        const bookingDetails = {};
+
+        if (!this.luisRecognizer.isConfigured) {
+            // LUIS is not configured, we just run the BookingDialog path.
+            return await stepContext.beginDialog('bookingDialog', bookingDetails);
+        }
+
+        // Call LUIS and gather any potential booking details. (Note the TurnContext has the response to the prompt)
+        const luisResult = await this.luisRecognizer.executeLuisQuery(stepContext.context);
+        switch (LuisRecognizer.topIntent(luisResult)) {
+        case 'BookFlight': {
+            // Extract the values for the composite entities from the LUIS result.
+            const fromEntities = this.luisRecognizer.getFromEntities(luisResult);
+            const toEntities = this.luisRecognizer.getToEntities(luisResult);
+
+            // Show a warning for Origin and Destination if we can't resolve them.
+            await this.showWarningForUnsupportedCities(stepContext.context, fromEntities, toEntities);
+
+            // Initialize BookingDetails with any entities we may have found in the response.
+            bookingDetails.destination = toEntities.airport;
+            bookingDetails.origin = fromEntities.airport;
+            bookingDetails.travelDate = this.luisRecognizer.getTravelDate(luisResult);
+            console.log('LUIS extracted these booking details:', JSON.stringify(bookingDetails));
+
+            // Run the BookingDialog passing in whatever details we have from the LUIS call, it will fill out the remainder.
+            return await stepContext.beginDialog('bookingDialog', bookingDetails);
+        }
```

Comment:
These brackets are not needed.  Please limit to updates required to implement telemetry.

```suggestion
```